src/consensus/aft/async_execution_coordinator.h

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ds/thread_messaging.h"
+
+namespace aft
+{
+  enum class AsyncExecutionResult
+  {
+    PENDING,
+    COMPLETE
+  };
+
+  class AsyncExecutionCoordinator
```

Comment:
Could this be AsyncExecutor? This is just conducting async execution, there is no coordination with any other component is there?